src/cpp/src/tokenizer.cpp

```diff
@@ -59,6 +59,18 @@ constexpr char bos_token_key_name[] = "bos_token";
 constexpr char eos_token_key_name[] = "eos_token";
 constexpr char pad_token_key_name[] = "pad_token";
 
+static std::shared_ptr<ov::Core> core;
+static ov::Core get_core_singleton() {
+    if (core) {
+        return *core;
```

Comment:
I didn't have any specific intentions to put it outside, just placed without trying to align with the original singleton. 

Anyway, i changed solution a bit, instead of shared_ptr i used bool flag and this time declared all of the inside of the func. Please take a look